### DISCUSSION TITLE
[Feature request]: Reduce the time it takes to connect to live view.

### DISCUSSION THREAD

**Discussion by xiaomizhouya** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#2023] [Q&A]
### Please confirm you have already done the following

- [X] I have searched the repository for related/existing feature requests
- [X] I have all the details the issue requires

### Describe the feature

I would like to know the role of **Nonce** in communicating with the **turn server**. Assuming the following scenario, the **fake master** allocates a request to the turn server and obtains a Nonce. If this Nonce is sent to the **real master** and then the **real master** sends a permission request to the turn server with this Nonce, can it be successful? Why did the turn server return an error of **438 (stale nonce)** with a new nonce in my test, and when **the real master** requested permission again with the new nonce, the turn server returned **401 (unauthenticated)**. Does this mean that the turn server distinguishes between **fake master** and **real master**, and based on what?

I would like to know if the **fake master** can help the **real master** with some tasks, such as connecting to the **signaling server**, **sdp answer**，**receive remote candidate**，**send local candidate**，Is it feasible? Is there a timeout issue?

### Use Case

I'm trying to reduce the time it takes for the **real master** and **peer** to connect by helping the real master do something with **fake master**.

**Comment by disa6302** [comment]
@xiaomizhouya ,

The nonce is a unique value generated by the TURN server and used once. When a client sends a request, it includes this nonce. If an attacker tries to replay a previously captured request, the nonce will have changed, and the server will recognize it as invalid. Nonces are used along with the username and password to authenticate requests. The server generates the nonce, and the client must include this nonce in its subsequent requests. This helps ensure that the request is coming from a legitimate client. The allocate request is what sends the nonce back to the client. The first time, the TURN server responds with a 401 Unauthorized status. This response includes a nonce and realm attribute. The nonce is a unique value generated by the server, and the realm is a string used for the authentication domain.

There is no way to get around this or cache the nonce in anyway. The best way to reduce TURN related latency would be to ensure the region being used in the device is as close to the TURN server region as possible.
